pkg/schema/schema.go

```diff
@@ -381,34 +456,44 @@ func removeDefaultHelper(inputdef cue.Value, input cue.Value) (cue.Value, bool,
 		if isCueValueEqual(inputdef, input) {
 			return rv, true, nil
 		}
-		ele := inputdef.LookupPath(cue.MakePath(cue.AnyIndex))
-		if ele.IncompleteKind() == cue.BottomKind {
-			return rv, true, nil
-		}
 
-		iter, err := input.List()
-		if err != nil {
-			return rv, true, nil
-		}
+		// take every element of the list
+		ele := inputdef.LookupPath(cue.MakePath(cue.AnyIndex))
+		// if input is not a concrete list, we must have list elements exist to be used to trim defaults
+		if ele.Exists() {
+			if ele.IncompleteKind() == cue.BottomKind {
+				return rv, true, nil
+			}
 
-		// The following code is workaround since today overwrite list element doesn't work
-		var iterlist []string
-		for iter.Next() {
-			re, isEqual, err := removeDefaultHelper(ele, iter.Value())
-			if err == nil && !isEqual {
-				reString, err := convertCUEValueToString(re)
+			iter, err := input.List()
+			if err != nil {
+				return rv, true, nil
+			}
+			var iterlist []string
+			for iter.Next() {
+				ref, err := getBranch(ele, iter.Value())
 				if err != nil {
 					return rv, true, nil
 				}
-				iterlist = append(iterlist, reString)
+				re, isEqual, err := removeDefaultHelper(ref, iter.Value())
+				if err == nil && !isEqual {
+					reString, err := convertCUEValueToString(re)
+					if err != nil {
+						return rv, true, nil
+					}
+					iterlist = append(iterlist, reString)
+				}
 			}
+			iterlistContent := fmt.Sprintf("[%s]", strings.Join(iterlist, ","))
```

Comment:
I am not sure now the list fill path index is working, it was not working at all, that is why i have to generate the string and build the entire list again to fill the list instead of list element.

Reply:
i have also found lists quite frustrating to work with 😢

Reply:
This func may help https://pkg.go.dev/cuelang.org/go@v0.4.0/cue#Context.NewList

Reply:
going to fix it in the pr of update 0.4.0

Reply:
good, we need that PR for multiple reasons now, anyway